From 6b2b5c625798facae8f0fc3cad2f37d09123e5e7 Mon Sep 17 00:00:00 2001
From: Icenowy Zheng <icenowy@aosc.io>
Date: Fri, 5 May 2017 16:19:48 +0800
Subject: [PATCH 94/94] drm/sun4i: fix TCON channel 1 vtotal value when
 interlaced

When it's interlaced the TCON vtotal value should be mode->vtotal, which
is the original vtotal value, rather than mode->crtc_vtotal, which is
only half the vtotal value.

Fix this value.

Signed-off-by: Icenowy Zheng <icenowy@aosc.io>
---
 drivers/gpu/drm/sun4i/sun4i_tcon.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/drivers/gpu/drm/sun4i/sun4i_tcon.c b/drivers/gpu/drm/sun4i/sun4i_tcon.c
index 2d9d7a708503..e4b85a63e1f3 100644
--- a/drivers/gpu/drm/sun4i/sun4i_tcon.c
+++ b/drivers/gpu/drm/sun4i/sun4i_tcon.c
@@ -259,7 +259,7 @@ void sun4i_tcon1_mode_set(struct sun4i_tcon *tcon,
 		     SUN4I_TCON1_BASIC3_H_BACKPORCH(bp));
 
 	bp = mode->crtc_vtotal - mode->crtc_vsync_start;
-	val = mode->crtc_vtotal;
+	val = mode->vtotal;
 	if (!(mode->flags & DRM_MODE_FLAG_INTERLACE))
 		val = val * 2;
 	DRM_DEBUG_DRIVER("Setting vertical total %d, backporch %d\n",
-- 
2.12.2

